_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-podcast-player.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Podcast Player API
+ *
+ * @package Jetpack
+ * @since 8.4.0
+ */
+
+/**
+ * Fetch podcast feeds and parse data for the Podcast Player block.
+ *
+ * @since 8.4.0
+ */
+class WPCOM_REST_API_V2_Endpoint_Podcast_Player extends WP_REST_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'podcast-player';
+		// This endpoint *does not* need to connect directly to Jetpack sites.
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Register the route.
+	 */
+	public function register_routes() {
+		// GET /sites/<blog_id>/podcast-player - Returns feed data.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'  => WP_REST_Server::READABLE,
+					'callback' => array( $this, 'get_player_data' ),
+					'args'     => array(
+						'url' => array(
+							'description'       => __( 'The Podcast RSS feed URL.', 'jetpack' ),
+							'type'              => 'string',
+							'required'          => 'true',
+							'validate_callback' => function ( $param ) {
+								return wp_http_validate_url( $param );
+							},
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Retreives data needed to display a podcast player from RSS feed.
+	 *
+	 * @param WP_REST_Request $request The REST API request data.
+	 * @return WP_REST_Response The REST API response.
+	 */
+	public function get_player_data( $request ) {
+		return rest_ensure_response(
+			array(
+				'tracks' => \Automattic\Jetpack\Extensions\Podcast_Player\get_track_list( $request['url'] ),
+				'cover'  => '', // TODO: parse podcast cover image.
+			)
+		);
```

Comment:
I wonder if we should cache the result in a transient. The endpoint will be publicly available for every Jetpack site out there, without any restrictions, so I wonder if there is something we could do to avoid abuse or to avoid having a site go down because of too many requests.

Reply:
How about we cache the response in a transient for 1hr. Would that help?

Reply:
I agree caching would be good. I think we need to make it on a different level than here so the data we show on the frontend always matches the editor (API). If we only cache the API, the backend might get more fresh data, causing a difference between them

Reply:
I've added a task to our board.